### PR TITLE
Chirality support in Developer touch layout editor

### PR DESCRIPTION
**Pull request by mcdurdin**

### CODE REVIEW

File: windows/src/developer/TIKE/oskbuilder/TouchLayoutDefinitions.pas [hunk 1]
```diff
@@ -92,15 +92,42 @@   TJSONDef = record
...
     'shift',
     'ctrl',
     'alt',
     'ctrlalt',
     'ctrlshift',
     'altshift',
-    'ctrlaltshift'
+    'ctrlaltshift',
+
+    // 10.0 predefined layer names
```
Comment (jahorton) [inline windows/src/developer/TIKE/oskbuilder/TouchLayoutDefinitions.pas]: I assume these are mutually exclusive sets, and that 'shift' and 'default' will still be selectable for chiral keyboards?

File: windows/src/developer/TIKE/xml/layoutbuilder/builder.js [hunk 2]
```diff
@@ -1408,8 +1499,8 @@ $(function () {
     buttons: {
       "OK": function () {
         var newLayerName = $('#layerName').val();
-        if (!newLayerName.match(/^[a-zA-Z0-9_]+$/)) {
-          alert('Layer name must contain only alphanumerics and underscore.');
+        if (!newLayerName.match(/^[a-zA-Z0-9_-]+$/)) {
+          alert('Layer name must contain only alphanumerics, hyphen underscore.');
```
Comment (jahorton) [inline windows/src/developer/TIKE/xml/layoutbuilder/builder.js]: `, hyphen **and** underscore.`

File: windows/src/developer/TIKE/xml/layoutbuilder/physical-keyboard-template.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+  "row": [
```
Comment (jahorton) [inline windows/src/developer/TIKE/xml/layoutbuilder/physical-keyboard-template.js]: What does this file correspond to?  It doesn't have both the left and right versions of the chiral modifiers, so I'm curious.

File: windows/src/developer/TIKE/oskbuilder/TouchLayoutDefinitions.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/oskbuilder/TouchLayoutDefinitions.pas]: Yes; these are used only for merging desktop visual keyboards with touch keyboards (during an import). The legacy names are left for existing keyboards

File: windows/src/developer/TIKE/xml/layoutbuilder/physical-keyboard-template.js (hunk 3 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/xml/layoutbuilder/physical-keyboard-template.js]: This is a template of the base physical layout used during import from a KVK file to a touch layout. The modifiers that are used are loaded as longpress options under the LControl key. See `TframeTouchLayoutBuilder.SetupModifierKeysForImportedLayout`

**Pull request merged**